File: pset1/cash/cash.c
#include <cs50.h>
#include <stdio.h>
#include <math.h>

int main(void)
{
    float dollars = 0;
    int steps = 0;
    do
    {
        dollars = get_float("Please the change owed ! "); //takes the input from user
    }
    while (dollars < 0) ;
    int cents = round(dollars * 100); //chane it all to cents
    while (cents >= 25) // calculate the number of spent quarters
    {
        cents -= 25;
        steps++;
    }
    while (cents >= 10) // calculate the number of spent dimes
    {
        cents -= 10;
        steps++;
    }
    while (cents >= 5)  //  calculate the number of spent nickels
    {
        cents -= 5;
        steps++;
    }
    while (cents >= 1)  //  calculate the number of spent pennies
    {
        cents--;
        steps++;
    }
    
    
    printf("%i\n", steps);
}

File: pset6/cash/cash.py

while True:
    try:
        val = float(input("Enter kosm change : "))    
        assert val >= 0
    except AssertionError:
        continue
    except ValueError:
        continue
    else:
        break 
    
cents = round(float(val) * 100)
steps = 0
while cents >= 25:
    cents -= 25
    steps += 1
while cents >= 10:
    cents -= 10
    steps += 1
while cents >= 5:
    cents -= 5
    steps += 1
while cents >= 1:
    cents -= 1
    steps += 1
print(steps, end="")